Tools/Parser/UnicodeData.cpp

```diff
@@ -69,6 +71,19 @@ UnicodeData::UnicodeData(const string &directory) :
             m_lastCodePoint = codePoint;
         }
 
+        std::string characterName;
+        getField(m_data, (size_t)offset, 1, characterName);
```

Comment:
Since the field will return whole data such as `<Non Private Use High Surrogate, First>`, it makes sense to fix the implementation of `getCharacterName` function as well.

Reply:
Method `getCharacterName` is actually not used anywhere in SheenBidi. There is no simple fix to return the correct character name for the individual code points of code point ranges, unless you interpret the comments for the related ranges in the Unicode data file `DerivedGeneralCategory.txt`.

For your example `<Non Private Use High Surrogate, First>..<Non Private Use High Surrogate, Last>` the correct names would be `<surrogate-D800>..<surrogate-DFFF>`. The previous implementation would have returned empty names for all code points between the first and last code point of a range, and incorrect names for the first and the last code point of a range. Now the entry of the last code point is used for all code poins of a range except for the first code point. Yes, it's not better, but it really doesn't matter for the purpose of generating a table for the general category.